### ISSUE TITLE
@named()

### ISSUE THREAD

**Issue by Sergey26Tarasenko** [fluttercommunity/get_it#10]
Hi! It's very nice library. Could you add `@named()` function? Thanks!

Example

`injector.registerLazySingleton<Client>(() => Client(), name: "Client1");
  injector.registerLazySingleton<Client>(() => Client(), name: "Client2");
  
  final client1 = injector.get<Client>("Client1");
  final client2 = injector.get<Client>("Client2");`

**Comment by escamoteur** [comment]
Hmm, not sure if I like it here. @anaisbetts what do you think?

**Comment by Sergey26Tarasenko** [comment]
for example:

need to add some sharedpreferance objects, but different by key name.

SharedPreference sp1 = SharedPreference(Name1, mode);
SharedPreference sp2 =
SharedPreference(Name2, mode);

**Comment by anaisbetts** [comment]
I think it's a good idea, because otherwise you end up having to make bogus types for silly reasons:

```
// The type here is `string`, a very common type!
locator.registerSingleton("http://the/url", name: "ServiceURL");
```

vs:

```
// ugh ugh ugh
class ServiceUrlContainer { ... }
locator.registerSingleton(ServiceUrlContainer("http://the/url"));
```

**Comment by lukepighetti** [comment]
New to service locators but that seems like an anti-pattern for `get_it` to store strings

**Comment by anaisbetts** [comment]
@lukepighetti Probably, there are more compelling examples though of objects that are difficult to differentiate solely on type. For example, if you have a `DiskCache` interface, you might want the `InUserProfile` and `TemporaryCached` instances, even though these are implemented by the same type

**Comment by escamoteur** [comment]
Implemented in V1.1.0
